llama_stack/distribution/resolver.py

```diff
@@ -243,7 +245,10 @@ def sort_providers_by_deps(
 
 
 async def instantiate_providers(
-    sorted_providers: list[tuple[str, ProviderWithSpec]], router_apis: set[Api], dist_registry: DistributionRegistry
+    sorted_providers: list[tuple[str, ProviderWithSpec]],
+    router_apis: set[Api],
+    dist_registry: DistributionRegistry,
+    resource_attributes: ResourceAccessAttributes,
```

Comment:
I think this argument should just be the `access_rules` -- i.e., it represents the data, not any behavior. the behavior itself doesn't even need a class (ResourceAccessAttributes) just a free floating function `check_access(user_abac, rules)` will do